### PR TITLE
☁️ 🐚 Migrate Azure Cloud Shell feature

### PR DESCRIPTION
**Pull request by alexweininger**
Purpose: In order to deprecate Azure Account, we need move the Azure Cloud Shell feature to the Azure Resources extension.

I will be making updates to the code later but wanted to perform an initial migration without changing the code too much.

* Removed webworker webpack config. We no longer ship a web version of Azure Resources, so it should've been removed a while ago. This config is not compatible with the Azure Cloud Shell feature.
* My process was copying over the files and adding comments. I tried to keep my changes minimal.
* Replaced the reliance on the Azure Account auth with our subscription provider. If a user has multiple tenants, we will prompt the user to select one every time they launch cloud shell. I think we can improve this later, but it's fine for now.
* The feature relies on `node-fetch` and `request-promise`, in a future update I hope to replace both of these with built-in fetch.

This feature is fairly complex and does a lot of out of the ordinary stuff to make it work. Please ask questions. I'm tentatively planning to write a document in the repo explaining the workings of the feature.

edit: Also I purposely left two prominent bugs in the code from AA. I will fix those in follow up PRs. The fixes for those are in https://github.com/microsoft/vscode-azureresourcegroups/pull/851, so use that for testing.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -626,13 +644,17 @@
     "dependencies": {
         "@azure/arm-resources": "^5.2.0",
         "@azure/arm-resources-profile-2020-09-01-hybrid": "^2.1.0",
+        "@azure/ms-rest-js": "^2.7.0",
         "@microsoft/vscode-azext-azureauth": "^2.3.0",
         "@microsoft/vscode-azext-azureutils": "^2.0.0",
         "@microsoft/vscode-azext-utils": "^2.4.0",
         "buffer": "^6.0.3",
         "jsonc-parser": "^2.2.1",
+        "node-fetch": "^3.3.2",
```
Comment (nturinski) [inline package.json:653]: Shouldn't this be native?

File: tsconfig.json [hunk 2]
```diff
@@ -4,14 +4,16 @@
         "target": "es6",
         "outDir": "out",
         "lib": [
-            "es6"
+            "es6",
+            "DOM"
```
Comment (nturinski) [inline tsconfig.json:8]: 👀

File: webpack.config.js [hunk 3]
```diff
@@ -44,18 +30,18 @@ const webConfig = dev.getDefaultWebpackConfig({
         '../build/default/validation': 'commonjs ../build/default/validation',
         '../build/Release/bufferutil': 'commonjs ../build/Release/bufferutil',
         '../build/default/bufferutil': 'commonjs ../build/default/bufferutil',
+
+        // required for cloud shell feature
+        bufferutil: 'commonjs bufferutil',
```
Comment (nturinski) [inline webpack.config.js:35]: This is the scary stuff 😬

**Review by nturinski — APPROVED** [review]
Don't see anything obviously wrong that stands out to me.

Only thing that I don't see is the equivalent of `uploadFile` being migrated over.
https://github.com/microsoft/vscode-azure-account/pull/958/files#diff-04bba6a35cad1c794cbbe677678a51de13441b7a6ee8592b7b50be1f05c6f626L112

**Comment by alexweininger** [comment]
> Only thing that I don't see is the equivalent of `uploadFile` being migrated over.
https://github.com/microsoft/vscode-azure-account/pull/958/files#diff-04bba6a35cad1c794cbbe677678a51de13441b7a6ee8592b7b50be1f05c6f626L112

https://github.com/microsoft/vscode-azureresourcegroups/pull/852

File: tsconfig.json (hunk 2 above)
Comment (alexweininger) [reply tsconfig.json:8]: I know. I'm not happy we have to add that but I hope to be able to remove it once I move this whole feature onto built-in fetch.

File: webpack.config.js (hunk 3 above)
Comment (alexweininger) [reply webpack.config.js:35]: Honestly I don't know what this does

Comment (nturinski) [reply webpack.config.js:35]: lmao I'm pretty sure it's polyfilling. Probably has to do with cloud shell stuff running on its own node process.

File: src/cloudConsole/cloudShellChildProcess/cloudConsoleLauncher.ts [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    };
+}
+
+let resizeToken = {};
+async function resize(accessToken: string, terminalUri: string) {
+    const token = resizeToken = {};
+    await delay(300);
+
+    for (let i = 0; i < 10; i++) {
+        if (token !== resizeToken) {
```
Comment (MicroFish91) [inline src/cloudConsole/cloudShellChildProcess/cloudConsoleLauncher.ts:54]: Q: Is it possible for `token` to not equal `resizeToken` here?

**Pull request merged**